### PR TITLE
use the variable already exist

### PR DESCRIPTION
**Pull request by getong**

### CODE REVIEW

**Comment by mrallen1** [comment]
Hi, appreciate the PR, but it looks like there's some problems running your tests in Travis. See for example: https://travis-ci.org/erlang-lager/lager/jobs/427249681

Can you take a look and address those please?

**Comment by getong** [comment]
I will.

File: test/lager_test_function_transform.erl [hunk 1]
```diff
@@ -122,10 +124,11 @@ transform_function_test_() ->
           ?assertEqual(1, lager_test_backend:count()),
           {_Level, _Time, _Message, Metadata} = lager_test_backend:pop(),
           Function = proplists:get_value(returns_dynamic_emit, Metadata),
-          ?assert(Function() < Function()),
-          ?assert(Function() < Function()),
-          ?assert(Function() < Function()),
-          ?assert(Function() < Function()),
+                ct:pal("Function:~p~n", [Function]),
```
Comment (mrallen1) [inline test/lager_test_function_transform.erl]: Why did all of this get changed? To emit log messages in eunit, you need to use the `?debugFmt()` macro not ct:pal :)

Comment (getong) [reply test/lager_test_function_transform.erl]: I am sorry, this is just my test code, I will remove it.

**Comment by getong** [comment]
@mrallen1 
The pr is finally ok, you can check the code.

File: test/lager_test_backend.erl [hunk 2]
```diff
@@ -1113,9 +1113,11 @@ error_logger_redirect_setup() ->
     application:load(lager),
     application:set_env(lager, error_logger_redirect, true),
     application:set_env(lager, handlers, [{?MODULE, info}]),
+    application:set_env(lager, suppress_supervisor_start_stop, false),
+    application:set_env(lager, suppress_application_start_stop, false),
     lager:start(),
     lager:log(error, self(), "flush flush"),
-    timer:sleep(100),
+    timer:sleep(1000),
```
Comment (mrallen1) [inline test/lager_test_backend.erl]: It was a bad idea to introduce sleeps in these functions. There has to be a cleaner way to signal that we're ready to proceed with tests.  I think the way to do that is to have the `flush` handler emit a message that the flush is completed. But I guess that could be a new enhancement tackled down the road.

I'm thinking like 

```erlang
   Ref = make_ref(),
   gen_event:call(lager_event, ?MODULE, {flush, Ref, self()}),
   receive
       {Ref, flush_completed} -> lager_event
   after ?TEST_TIMEOUT ->
        erlang:error(test_timeout)
   end.
```

I don't know. Maybe that's a terrible idea but there's got to be something better than a sleep.

**Comment by mrallen1** [comment]
Also it would be cool if you'd squash some of the commits you ended up abandoning. :)

**Comment by mrallen1** [comment]
Not sure if you intended to squash commits but they're not squashed. :)

**Comment by getong** [comment]
I recheck the commits, only the last five commits can be squashed into one commit. Or you mean to squash all commits into one commit?

**Comment by getong** [comment]
You can also cherry pick some commits and squash them.

**Comment by michaelklishin** [comment]
@getong some maintainers prefer all commits in a PR to be squashed into one.

**Comment by mrallen1** [comment]
In this case, yes, it would be nice to have them squished into a commit.

**Comment by getong** [comment]
ok, I will do it tonight.

**Comment by getong** [comment]
done.

**Pull request merged**